### PR TITLE
Clarify docs for ALTER TABLE EXECUTE

### PR DESCRIPTION
**Pull request by sheajamba**
## Description

This PR adds some clarifications to what procedures are available when running `ALTER TABLE EXECUTE`. 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/sql/alter-table.md [hunk 1]
```diff
@@ -54,15 +54,20 @@ per-connector basis.
...
+Executable commands are contributed by connectors, for example, to collapse
+files in a table that are over 128 megabytes in size with the `optimize` command
+from the [Hive](hive-alter-table-execute), [Delta
 Lake](delta-lake-alter-table-execute) , and
-[Iceberg](iceberg-alter-table-execute) connectors:
+[Iceberg](iceberg-alter-table-execute) connectors, you can run the following
+query:
 
 ```
 ALTER TABLE hive.schema.test_table EXECUTE optimize(file_size_threshold => '128MB')
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:65]: change hive.schema to example.test and explain in the opening sentence

File: docs/src/main/sphinx/sql/alter-table.md [hunk 2]
```diff
@@ -54,15 +54,20 @@ per-connector basis.
...
 ```
 
+Refer to individual [connector documentation](../connector.md) for more
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:68]:
```suggestion
Refer to individual [connector documentation](connector) for more
```

Comment (mosabua) [reply docs/src/main/sphinx/sql/alter-table.md:68]: Also maybe this sentence is redundant anyway .. since you are already linking above

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/sql/alter-table.md [hunk 3]
```diff
@@ -54,15 +54,20 @@ per-connector basis.
 You can use the `=>` operator for passing named parameter values.
 The left side is the name of the parameter, the right side is the value being passed.
 
-Collapse files in a table that are over 128 megabytes in size with the `optimize`
-procedure from the [Hive](hive-alter-table-execute), [Delta
+Executable commands are contributed by connectors, for example, to collapse
+files in a table that are over 128 megabytes in size with the `optimize` command
```
Comment (hashhar) [inline docs/src/main/sphinx/sql/alter-table.md:58]:
```suggestion
Executable commands are contributed by connectors, for example, to merge
files in a table that are over 128 megabytes in size with the `optimize` command
```

(It does more than just merging files in Iceberg though)

**Review by hashhar — APPROVED** [review]

File: docs/src/main/sphinx/sql/alter-table.md [hunk 4]
```diff
@@ -54,13 +54,14 @@ per-connector basis.
 You can use the `=>` operator for passing named parameter values.
 The left side is the name of the parameter, the right side is the value being passed.
 
-Collapse files in a table that are over 128 megabytes in size with the `optimize`
-procedure from the [Hive](hive-alter-table-execute), [Delta
-Lake](delta-lake-alter-table-execute) , and
-[Iceberg](iceberg-alter-table-execute) connectors:
+Executable commands are contributed by connectors, for example, to merge files
+in a table called `test_table` that is over 128 megabytes in size within the
```
Comment (findinpath) [inline docs/src/main/sphinx/sql/alter-table.md:58]: Optimize is used to "merge" files which are under the specified size, not over the specified size

See https://trino.io/docs/current/connector/iceberg.html#optimize

> The following statement merges files in a table that are under 128 megabytes in size:
> 
> `ALTER TABLE test_table EXECUTE optimize(file_size_threshold => '128MB')`

File: docs/src/main/sphinx/sql/alter-table.md [hunk 5]
```diff
@@ -54,13 +54,14 @@ per-connector basis.
 You can use the `=>` operator for passing named parameter values.
 The left side is the name of the parameter, the right side is the value being passed.
 
-Collapse files in a table that are over 128 megabytes in size with the `optimize`
-procedure from the [Hive](hive-alter-table-execute), [Delta
-Lake](delta-lake-alter-table-execute) , and
-[Iceberg](iceberg-alter-table-execute) connectors:
+Executable commands are contributed by connectors, for example, to merge files
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:57]:
```suggestion
Executable commands are contributed by connectors. For example, to merge files
```

File: docs/src/main/sphinx/sql/alter-table.md (hunk 4 above)
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:58]: This has to be reworded ..

its not the table that is over 128mb .. its about merging all the many data files in that table that are smaller than 128mb into fewer files that are larger than 128mb

So the initial suggestion from @findinpath is probably better .. but also a bit terse and assuming a lot of knowledge. 

Lastly .. do you think 128mb file size is a reasonable example @findinpath ?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/sql/alter-table.md (hunk 4 above)
Comment (findinpath) [reply docs/src/main/sphinx/sql/alter-table.md:58]: > Lastly .. do you think 128mb file size is a reasonable example @findinpath ?

I'm missing field experience here.

@raunaqmorarka can you help on this one?

Comment (raunaqmorarka) [reply docs/src/main/sphinx/sql/alter-table.md:58]: We should use an example where files are obviously small, like 16mb

Comment (hashhar) [reply docs/src/main/sphinx/sql/alter-table.md:58]: 128 was chosen because people copy the examples and 128MB is ideal size for most parquet based tables.

Comment (mosabua) [reply docs/src/main/sphinx/sql/alter-table.md:58]: Great .. so we should explain that this command is useful when a user observes many small files in the 16mb range or smaller .. and it gets it to the ideal size of 128MB .. ping me to work on wording together if you like @sheajamba

File: docs/src/main/sphinx/sql/alter-table.md [hunk 6]
```diff
@@ -54,15 +54,19 @@ per-connector basis.
...
-procedure from the [Hive](hive-alter-table-execute), [Delta
-Lake](delta-lake-alter-table-execute) , and
-[Iceberg](iceberg-alter-table-execute) connectors:
+Executable commands are contributed by connectors, such as the `optimize`
+command provided by the [Hive](hive-alter-table-execute), [Delta
+Lake](delta-lake-alter-table-execute), and
+[Iceberg](iceberg-alter-table-execute) connectors. For example, a user with many
+small files in a table called `test_table` in the `example.test` schema, can use
+the `optimize` command to merge all files below the `file_size_threshold`
+parameter into fewer, but larger files:
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:63]:
```suggestion
value. The result is fewer, but larger files, which typically results in higher query performance on the data in the files:
```

File: docs/src/main/sphinx/sql/alter-table.md [hunk 7]
```diff
@@ -54,15 +54,19 @@ per-connector basis.
...
 
 ```
-ALTER TABLE hive.schema.test_table EXECUTE optimize(file_size_threshold => '128MB')
+ALTER TABLE example.test.test_table EXECUTE optimize(file_size_threshold => '16MB')
 ```
 
+
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/alter-table.md:69]: remove

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by mosabua — APPROVED** [review]
Looks good now.

**Pull request merged**